core/src/main/scala/org/broadinstitute/workbench/hamm/CostCalculator.scala

```diff
@@ -0,0 +1,95 @@
+package org.broadinstitute.workbench.hamm
+import java.time.{Duration, Instant}
+import java.util.concurrent.TimeUnit
+
+import cats.data.NonEmptyList
+import cats.implicits._
+import org.broadinstitute.workbench.hamm.pricing.{ComputePriceKey, PriceList, StoragePriceKey, UsageType}
+
+import scala.concurrent.duration.FiniteDuration
+
+
+object CostCalculator {
+
+  def getPriceOfWorkflow(callMetaDataJson: MetadataResponse, priceList: PriceList): Either[Throwable, Double] = {
+    val ls: List[Either[NonEmptyList[String], Double]] = callMetaDataJson.calls.map { call =>
+      getPriceOfCall(call, priceList, callMetaDataJson.startTime, callMetaDataJson.endTime).leftMap(NonEmptyList.one)
+    }
+
+    ls.parSequence.leftMap(errors => new Exception(errors.toList.toString)).map(_.sum)
+  }
+
+  private def getPriceOfCall(call: Call, priceList: PriceList, startTime: Instant, endTime: Instant): Either[String, Double] = {
+    for {
+      _ <- if (Status.terminalStatuses.contains(call.status)) Right(()) else Left(s"Call {name} status was ${call.status.asString}.") // not evaluating calls that are in flight
+      // ToDo: handle multiple disks in one call
```

Comment:
make jira tickets for all todos

Reply:
Yep! The one for this one is here: https://broadinstitute.atlassian.net/browse/GAWB-4076